app/components/pipeline/jobs/table/dataReloader.js

```diff
@@ -98,7 +98,11 @@ export default class DataReloader {
   }
 
   setNumBuilds(numBuilds) {
+    if (this.numBuilds === numBuilds) {
+      return;
+    }
     this.numBuilds = numBuilds;
+    this.fetchBuildsForJobs(this.jobIdsMatchingFilter).then(() => {});
```

Comment:
```suggestion
  async setNumBuilds(numBuilds) {
    if (this.numBuilds === numBuilds) {
      return;
    }
    this.numBuilds = numBuilds;
    await this.fetchBuildsForJobs(this.jobIdsMatchingFilter);
```
Let's make this an `async` function instead